src/server/app/shared/plugins/auth.js

```diff
@@ -39,38 +38,65 @@ async function getSecretFromUserId(userId) {
 }
 
 export async function generateToken(auth) {
-  let secret, payload, token;
   try {
-    secret = await getSecretFromUserId(auth.credentials.id);
-    // create payload
-    payload = {
-      ...auth.credentials,
-      dateGenerated: Date.now()
+    const secret = await getSecretFromUserId(auth.id);
+    const payloadAndHeaders = {
+      payload: {
+        username: auth.email,
+        dateGenerated: Date.now(),
+        expireAt: addHours(Date.now(), 7 * 24) // 1 week
+      },
+      header: { // must declare headers explicitly to add kid
+        typ: "JWT", // type of token
+        alg: "HS256", // algorithm used to encode token
+        kid: auth.email // key ID is hint to server on which secret to use
+      }
     }
     // encode token
-    token = await jwt.encode(secret, payload, (err, t)=> { return t; });
-    auth.credentials = payload;
-    auth.token = token;
-    return auth;
+    const token = await jwt.encode(secret, payloadAndHeaders, (err, t)=> { return t; });
+    return token;
   } catch (err) {
     if (Bounce.isSystem(err)) logErr("generateToken error: ", err.message || err);
     return new Error(err);
   }
 };
 
-export async function validateToken(username, token) {
+export async function validateToken(name, token) {
```

Comment:
Minor question, but why did we switch away from the `username` arg? It looks like we're still passing a `username`, unless I missed something?

Reply:
Now that I think about it, that could be confusing if we end up using the 'user' field in the User model, but I originally changed it for two reasons: 1) I ended up using the 'username' var in the decodedToken so I had to use something else for it here, and 2) We technically can do logins based on email OR phone number-- we haven't implemented the phone number but it's a field in the db.

Reply:
> I ended up using the 'username' var in the decodedToken so I had to use something else for it here

Sure that's fine!

> We technically can do logins based on email OR phone number-- we haven't implemented the phone number but it's a field in the db.

I would maybe suggest using the word `identifier` then, since that seems to suggest email or phone number more effectively than `name`. I would also say separately, though, that we should consider removing that phone number feature unless we truly think we need it. Given the fact that only members of these organizations will make accounts, I don't see how we would need it.